File: src/REXOS/HAL/ModuleTypeIdentifier.java
package HAL;

import org.json.JSONException;
import org.json.JSONObject;

import util.log.LogLevel;
import util.log.LogSection;
import util.log.Logger;
/**
 * This class provides a method of identifying module types
 * @author Tommas Bakker
 *
 */
public class ModuleTypeIdentifier {
	private String manufacturer;
	private String typeNumber;
	
	public ModuleTypeIdentifier(String manufacturer, String typeNumber){
		this.manufacturer = manufacturer;
		this.typeNumber = typeNumber;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (obj instanceof ModuleTypeIdentifier)
			return false;
		ModuleTypeIdentifier other = (ModuleTypeIdentifier) obj;
		if (manufacturer == null) {
			if (other.manufacturer != null)
				return false;
		} else if (!manufacturer.equals(other.manufacturer))
			return false;
		if (typeNumber == null) {
			if (other.typeNumber != null)
				return false;
		} else if (!typeNumber.equals(other.typeNumber))
			return false;
		return true;
	}

	public String getManufacturer(){
		return this.manufacturer;
	}
	public String getTypeNumber(){
		return this.typeNumber;
	}
	public JSONObject toJSON(){
		JSONObject moduleIdentifier = new JSONObject();
		try {
			moduleIdentifier.putOpt("manufacturer", manufacturer);
			moduleIdentifier.putOpt("typeNumber", typeNumber);
		} catch (JSONException ex) {
			Logger.log(LogSection.HAL, LogLevel.EMERGENCY, "Error occurred which is considered to be impossible", ex);
		}
		return moduleIdentifier;
	}
	
	@Override
	public String toString() {
		return "man " + manufacturer + " tNr " + typeNumber;
	}
	
}

File: src/REXOS/ROS/libraries/rexos_datatypes/src/ModuleTypeIdentifier.cpp
#include <rexos_datatypes/ModuleTypeIdentifier.h>

namespace rexos_datatypes{
	ModuleTypeIdentifier::ModuleTypeIdentifier(std::string manufacturer, std::string typeNumber) :
			manufacturer(manufacturer), typeNumber(typeNumber)
	{
		// nothing to do here
	}
	std::string ModuleTypeIdentifier::getManufacturer() const {
		return manufacturer;
	}
	std::string ModuleTypeIdentifier::getTypeNumber() const {
		return typeNumber;
	}
	std::ostream& operator<<(std::ostream& os, const ModuleTypeIdentifier& obj) {
		os << "manufacturer: " << obj.getManufacturer();
		os << "typeNumber: " << obj.getTypeNumber();
		return os;
	}
}